Content.Server/_CorvaxNext/Botany/Systems/PlantAnalyzerSystem.cs

```diff
@@ -0,0 +1,259 @@
+        {
+            if (!_cell.TryUseActivatableCharge(ent, user: args.User))
+                return;
+        }
+        if (args.Handled || args.Cancelled || args.Args.Target == null || !_cell.TryUseActivatableCharge(ent.Owner, user: args.User))
+            return;
+
+        _audio.PlayPvs(ent.Comp.ScanningEndSound, ent);
+
+        OpenUserInterface(args.User, ent);
+        UpdateScannedUser(ent, args.Args.Target.Value);
+
+        args.Handled = true;
+    }
+
+    private void OpenUserInterface(EntityUid user, EntityUid analyzer)
+    {
+        if (!TryComp<ActorComponent>(user, out var actor) || !_uiSystem.HasUi(analyzer, PlantAnalyzerUiKey.Key))
+            return;
+
+        _uiSystem.OpenUi(analyzer, PlantAnalyzerUiKey.Key, actor.PlayerSession);
+    }
+
+    public void UpdateScannedUser(Entity<PlantAnalyzerComponent> ent, EntityUid target)
+    {
+        if (!_uiSystem.HasUi(ent, PlantAnalyzerUiKey.Key))
+            return;
+
+        if (TryComp<SeedComponent>(target, out var seedComp))
+        {
+            if (seedComp.Seed != null)
+            {
+                var state = ObtainingGeneDataSeed(seedComp.Seed, target, false, ent.Comp.Settings.AdvancedScan);
+                _uiSystem.ServerSendUiMessage(ent.Owner, PlantAnalyzerUiKey.Key, state);
+            }
+            else if (seedComp.SeedId != null && _prototypeManager.TryIndex(seedComp.SeedId, out SeedPrototype? protoSeed))
+            {
+                var state = ObtainingGeneDataSeed(protoSeed, target, false, ent.Comp.Settings.AdvancedScan);
+                _uiSystem.ServerSendUiMessage(ent.Owner, PlantAnalyzerUiKey.Key, state);
+            }
+        }
+        else if (TryComp<PlantHolderComponent>(target, out var plantComp))
+        {
+            if (plantComp.Seed != null)
+            {
+                var state = ObtainingGeneDataSeed(plantComp.Seed, target, true, ent.Comp.Settings.AdvancedScan);
+                _uiSystem.ServerSendUiMessage(ent.Owner, PlantAnalyzerUiKey.Key, state);
+            }
+        }
+    }
+
+    /// <summary>
+    ///     Analysis of seed from prototype.
+    /// </summary>
+    public PlantAnalyzerScannedSeedPlantInformation ObtainingGeneDataSeed(SeedData seedData, EntityUid target, bool isTray, bool scanIsAdvanced)
+    {
+        // Get trickier fields first.
+        AnalyzerHarvestType harvestType = AnalyzerHarvestType.Unknown;
+        switch (seedData.HarvestRepeat)
+        {
+            case HarvestType.Repeat:
+                harvestType = AnalyzerHarvestType.Repeat;
+                break;
+            case HarvestType.NoRepeat:
+                harvestType = AnalyzerHarvestType.NoRepeat;
+                break;
+            case HarvestType.SelfHarvest:
+                harvestType = AnalyzerHarvestType.SelfHarvest;
+                break;
+            default:
+                break;
+        }
+
+        var mutationProtos = seedData.MutationPrototypes;
+        List<string> mutationStrings = new();
```

Comment:
```suggestion
        List<string> mutationStrings = [];
```
Ненавижу эту фигню, но IDE подсвечивает, что тут так надо.

Reply:
Оно пиздит, все начинает падать, а чатджипити говорит следующее:

В C# синтаксис `List<string> mutationStrings = [];` **не является валидным**. 

### Правильный синтаксис:
```csharp
List<string> mutationStrings = new();
```

---

### Почему так?
В C#, начиная с версии 9.0, введён **целевой тип вывода** (target-typed new expressions). Это позволяет указывать только `new()` вместо полного типа при инициализации объекта, если компилятор может вывести тип из контекста. 

Однако синтаксис `[]` для инициализации списков (как, например, в Python или JavaScript) в C# отсутствует.

---

### Альтернатива: Использование литералов коллекций
Если вы хотите сразу инициализировать список значениями, можно использовать **литералы коллекций**, начиная с версии C# 6.0:

```csharp
List<string> mutationStrings = new List<string> { "value1", "value2" };
```

В C# 10.0 и выше, когда тип можно вывести из контекста, также допустим вариант:

```csharp
var mutationStrings = new List<string> { "value1", "value2" };
```

---

### Рекомендация
Используйте:

```csharp
List<string> mutationStrings = new();
```

- Это современный, краткий и понятный синтаксис.
- Поддерживается в C# 9.0 и выше.
- Сохраняет строгую типизацию, позволяя компилятору выводить тип.